packages/shared/routes/setup/backup/views/BackupToFile.svelte

```diff
@@ -1,32 +1,47 @@
 <script lang="typescript">
+    import { createEventDispatcher } from 'svelte'
     import { Animation, Button, OnboardingLayout, Password, Spinner, Text } from 'shared/components'
     import { mobile } from '@core/app'
-    import { createEventDispatcher } from 'svelte'
+    import { mnemonic } from '@contexts/onboarding'
     import { Locale } from '@core/i18n'
+    import { backupInitialStronghold } from '@contexts/onboarding'
+    import { storeMnemonic } from '@core/profile-manager'
 
     export let locale: Locale
     export let strongholdPassword
     export let busy = false
 
     let confirmPassword
-    let skipping = false
+    let skipBackup = false
 
     const dispatch = createEventDispatcher()
 
-    $: valid = strongholdPassword === confirmPassword
+    $: isStrongholdPasswordValid = strongholdPassword === confirmPassword
+
+    function handleBackClick() {
+        dispatch('previous')
+    }
+
+    async function onboardingBackupFileFunction(_skipBackup: boolean = false): Promise<void> {
+        skipBackup = _skipBackup
 
-    function onSubmit() {
-        if (valid) {
-            skipping = false
+        await storeMnemonic($mnemonic.join(' '))
+        if (!_skipBackup) {
+            await backupInitialStronghold()
             dispatch('next')
+        } else {
+            dispatch('next', { skip: true })
```

Comment:
Maybe switch the if, else statement to remove the negation?

Reply:
Yeah :sweat_smile: